frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/create-local-volume-set.tsx

```diff
@@ -0,0 +1,256 @@
+import { NodesSelectionList } from './nodes-selection-list';
+import { RowUIDMap, LocalVolumeSetKind } from './types';
+import { getSelectedNodeUIDs } from './utils';
+import './create-local-volume-set.scss';
+
+const VOLUME_TYPE = {
+  SSD: {
+    title: 'SSD / NVMe',
+    property: 'Non Rotational',
+  },
+  HDD: {
+    title: 'HDD',
+    property: 'Rotational',
+  },
+};
+
+const VOLUME_MODE = {
+  Block: {
+    title: 'Block',
+  },
+  Filesystem: {
+    title: 'Filesystem',
+  },
+};
+
+const getDropdownItems = (obj: { [key: string]: { title: string } }) =>
+  Object.keys(obj).reduce((items, type) => {
+    items[type] = obj[type].title;
+    return items;
+  }, {});
+
+const volumeTypeDropdownItems = getDropdownItems(VOLUME_TYPE);
+const volumeModeDropdownItems = getDropdownItems(VOLUME_MODE);
+
+const CreateLocalVolumeSet: React.FC = withHandlePromise<CreateLocalVolumeSetProps>((props) => {
+  const { match, handlePromise, inProgress, errorMessage } = props;
+  const [volumeSetName, setVolumeSetName] = React.useState('');
+  const [storageClassName, setStorageClassName] = React.useState('');
+  const [showNodesList, setShowNodesList] = React.useState(false);
+  const [volumeType, setVolumeType] = React.useState('SSD');
+  const [volumeMode, setVolumeMode] = React.useState('Block');
+  const [maxVolumeLimit, setMaxVolumeLimit] = React.useState('');
+  const [rows, setRows] = React.useState<RowUIDMap>({});
+  const [allSelected, setAllSelected] = React.useState(null);
+
+  const { ns, appName } = match.params;
+  const modelName = LocalVolumeSetModel.label;
+
+  const toggleShowNodesList = () => {
+    setShowNodesList(!showNodesList);
+  };
+
+  const onSubmit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const requestData: LocalVolumeSetKind = {
+      apiVersion: LocalVolumeSetModel.apiVersion,
+      kind: LocalVolumeSetModel.kind,
+      metadata: { name: volumeSetName },
+      spec: {
+        storageClassName,
+        volumeMode,
+        deviceInclusionSpec: {
+          // Only Raw disk supported for 4.5
+          deviceTypes: ['RawDisk'],
+          deviceMechanicalProperty: VOLUME_TYPE[volumeType].property,
+        },
+      },
+    };
+    if (showNodesList) {
+      const selectedNodesUID = getSelectedNodeUIDs(rows);
+      const selectedNodes = selectedNodesUID.map((uid) => rows[uid].props.data.metadata.name);
+      requestData.spec.nodeSelector = {
+        nodeSelectorTerms: [
+          {
+            matchExpressions: [
+              { key: 'kubernetes.io/hostname', operator: 'In', values: [...selectedNodes] },
+            ],
+          },
+        ],
+      };
+    }
+
+    if (maxVolumeLimit) requestData.spec.maxDeviceCount = +maxVolumeLimit;
+    handlePromise(k8sCreate(LocalVolumeSetModel, requestData))
+      .then((resource) => {
+        history.push(resourceObjPath(resource, referenceFor(resource)));
+      })
+      .catch(() => null);
```

Comment:
```suggestion
```
No need for the catch statement

Reply:
This is enforced by linter

Reply:
It's not. `handlePromise` handles the catch part for you.

Reply:
Surely `handlePromise`  sets state and error message on catching error but it also returns a promise. Linters are dumb they dont know that, hence I need to put a dummy catch.
This will not break anything in our case because we have error handled in state.
https://github.com/openshift/console/blob/b270084647ae90a7a3e826e2a89c385d82de9769/frontend/public/components/utils/promise-component.tsx#L72